src/main/java/com/bobocode/petros/scaner/AnnotationDependencyClassScanner.java

```diff
@@ -1,17 +1,175 @@
+            throw new DefaultConstructorNotFoundException(aClass.getSimpleName());
+        }
+        return true;
+    }
+
+    private DependencyDefinition createInnerDependencyDefinition(Class<?> aClass) {
+        var definition = new DependencyDefinition();
+        definition.setName(firstCharacterToLowercase(aClass.getSimpleName()));
+        definition.setQualifiedName(getQualifiedNameFromPath(aClass));
+        return definition;
+    }
+
+    private List<Class<?>> getConstructorsParam(Class<?> aClass) {
+        return Arrays.stream(getDeclaredConstructors(aClass))
+                .flatMap(a -> Arrays.stream(a.getParameterTypes()))
+                .collect(toList());
+    }
+
+    private Stream<Path> getPathStream(Path path) {
+        try {
+            return Files.walk(path);
+        } catch (IOException e) {
+            LOG.error("Package {} unsuccessfully scanned", getSimpleClassPathName(path));
+            throw new NoSuchPackageFoundException(getSimpleClassPathName(path));
+        }
+    }
+
+    private String getQualifiedNameFromPath(Class<?> aClass) {
+        return aClass.getPackage().getName() + "." + aClass.getSimpleName();
+    }
+
+    private boolean hasDefaultConstructorByPath(Path path) {
+        var hasDefaultConstructor = Arrays.stream(getDeclaredConstructors(getClassByPath(path)))
+                .anyMatch(a -> a.getParameterCount() == COUNT_OF_PARAMETER_IN_CONSTRUCTOR);
+        if (!hasDefaultConstructor) {
+            LOG.error("Class {} doesn't have default constructor", getSimpleClassPathName(path));
+            throw new DefaultConstructorNotFoundException(getSimpleClassPathName(path));
+        }
+        return true;
+    }
+
+    private Class<?> getClassWhichMarkedAsDependency(Path path) {
+        return getClassByPath(path);
+    }
+
+    private boolean hasClassMarkedAsDependency(Path path) {
+        return getClassByPath(path)
+                .isAnnotationPresent(Dependency.class);
+    }
+
+    private Class<?> getClassByPath(Path path) {
+        try {
+            return Class.forName(getSimpleClassPathName(path));
+        } catch (ClassNotFoundException e) {
+            throw new DependencyClassNotFoundException(getSimpleClassPathName(path));
+        }
+    }
+
+    private static String getSimpleClassPathName(Path path) {
+        try {
+            return path.toString().replace("/", ".")
+                    .substring(path.toString().indexOf("com"))
+                    .replace(".class", "");
+        } catch (IndexOutOfBoundsException e) {
+            throw new NoSuchPathFoundException(path.toString());
+        }
+    }
+
+    private String firstCharacterToLowercase(String str) {
+        var strName = str.substring(0, 1).toLowerCase() + str.substring(1);
+        if (strName.substring(1, 2).equals(strName.substring(1, 2).toUpperCase())) {
```

Comment:
I can't understand what is going on here... As for me first line of that method is doing everything we need. Could you explain in few words why we need that?

Reply:
I almost made as Spring bean naming but i think need just delete second logic.
For instance how spring make naming of classes : 
**Case one** class with name "Class" -> his bean name "class" in spring context
**Case two** class with name "MYclass" -> his bean name "myclass" in spring context
**Case three** case with name "ASCclass" -> his bean name "ASCclass" in spring context
But I think it isn't important and we can delete this logic.

Reply:
Done!